File: 250207/요일 맞추기/guess-day-of-week.py
m1, d1, m2, d2 = map(int, input().split())
num_of_days = [0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

# 2011년 1월 1일 기준 계산
days1 = sum(num_of_days[:m1]) + d1
days2 = sum(num_of_days[:m2]) + d2

days = days2-days1
if days%7==0:
    print("Mon")
elif days%7==1:
    print("Tue")
elif days%7==2:
    print("Wed")
elif days%7==3:
    print("Thu")
elif days%7==4:
    print("Fri")
elif days%7==5:
    print("Sat")
elif days%7==6:
    print("Sun")

File: trail2/요일 맞추기/guess-day-of-week.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int m1 = sc.nextInt();
        int d1 = sc.nextInt();
        int m2 = sc.nextInt();
        int d2 = sc.nextInt();

        int[] day = new int[] {0,31,28,31,30,31,30,31,31,30,31,30,31};
        int day1 = 0;
        int day2 = 0;
        for (int i=1; i<=12; i++) {
            if (i < m1) {
                day1 += day[i];
            }
            if (i < m2) {
                day2 += day[i];
            }
        }
        day1 += d1;
        day2 += d2;

        int diff = day2-day1;
        diff %= 7;
        if (diff < 0) {
            diff += 7;
        }

        String[] d = new String[] {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
        System.out.println(d[diff]);
    }
}